src/FSharpLint.Core/Application/Lint.fsi

```diff
@@ -119,11 +119,27 @@ module Lint =
         member TryGetSuccess : byref<Suggestion.LintWarning list> -> bool
         member TryGetFailure : byref<LintFailure> -> bool
 
+    type RunAstNodeRulesConfig = 
+        { Rules: RuleMetadata<AstNodeRuleConfig> []
+          GlobalConfig: Rules.GlobalRuleConfig
+          TypeCheckResults: FSharpCheckFileResults option
+          FilePath: string
+          FileContent: string
+          Lines: string []
+          SyntaxArray: AbstractSyntaxArray.Node array }
+
     /// Runs all rules which take a node of the AST as input.
-    val runAstNodeRules : RuleMetadata<AstNodeRuleConfig> [] -> Rules.GlobalRuleConfig -> FSharpCheckFileResults option -> string -> string -> string [] -> AbstractSyntaxArray.Node [] -> Suggestion.LintWarning [] * Context
+    val runAstNodeRules : RunAstNodeRulesConfig -> Suggestion.LintWarning array * Context
```

Comment:
@Mersho there isn't (yet) any rule that is forcing you to convert `[]` to `array` so please undo this change (here and in other places where you do the same replacement)